source/implementations/windows/Meadow.Windows/WindowsNetworkAdapterCollection.cs

```diff
@@ -135,7 +136,23 @@ private List<WifiNetwork> Scan()
 
     public AntennaType CurrentAntenna => throw new NotImplementedException();
 
-    public override bool IsConnected => throw new NotImplementedException();
+    public override bool IsConnected => IsInternetAvailable();
+
+    private bool IsInternetAvailable()

```

Comment:
This doesn't feel true.  That only says the network interface is up.  "Internet connected" implies a connection, and routability, to public sites.  If you can ping, say, 8.8.8.8, it's probably "internet available"

Reply:
oh yes, good idea.

Reply:
Logic changed, take a look when you get a chance @ctacke